src/include/sof/schedule.h

```diff
@@ -25,113 +25,98 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *
- * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ * Author: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>
  */
 
-#ifndef __INCLUDE_SOF_SCHEDULE_H__
-#define __INCLUDE_SOF_SCHEDULE_H__
+/* Generic schedule api header */
+#ifndef __INCLUDE_SOF_SCHEDULER_H__
+#define __INCLUDE_SOF_SCHEDULER_H__
 
 #include <stdint.h>
-#include <stddef.h>
-#include <errno.h>
-#include <sof/sof.h>
-#include <sof/lock.h>
 #include <sof/list.h>
-#include <sof/work.h>
-#include <sof/wait.h>
 
-/* schedule tracing */
-#define trace_schedule(format, ...) \
-	trace_event(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
-
-#define trace_schedule_error(format, ...) \
-	trace_error(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
-
-#define tracev_schedule(format, ...) \
-	tracev_event(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
+enum {
+	SOF_EDF_TASK = 0,
+	SOF_LOW_LATENCY_TASK,
+	SOF_TASKS_COUNT
+};
 
-struct schedule_data;
-struct sof;
+#define SOF_TASK_PRI_0		0	/* priority level 0 (lowest) */
+#define SOF_TASK_PRI_1		1	/* priority level 1 */
+#define SOF_TASK_PRI_2		2	/* priority level 2 */
+#define SOF_TASK_PRI_3		3	/* priority level 3 */
+#define SOF_TASK_PRI_4		4	/* priority level 4 */
+#define SOF_TASK_PRI_5		5	/* priority level 5 */
+#define SOF_TASK_PRI_6		6	/* priority level 6 */
+#define SOF_TASK_PRI_7		7	/* priority level 7 */
+#define SOF_TASK_PRI_8		8	/* priority level 8 */
+#define SOF_TASK_PRI_9		9	/* priority level 9 (highest) */
+#define SOF_TASK_PIORITIES	10	/* range of priorities (0-9) */
```

Comment:
I'd just use something like `#define SOF_TASK_PRI_MAX SOF_TASK_PRI_9`, naming it as `PRIORITIES` makes it look like array/list

Reply:
I've just rename it to ```#define SOF_TASK_PRI_COUNT	10```. We need amout of priorities for:
```
struct ll_queue {
	struct list_item work[LL_PRIORITIES];	/* list of ll per priority */
```
```
#define LL_PRIORITIES	SOF_TASK_PRI_COUNT
```

Reply:
This would be a lot nicer to have a single list of tasks rather than this array per priority. It's simple and fast to traverse.